### PR TITLE
Refresh expired token

### LINKED ISSUE #67: Expired authentication tokens don't always pop an error and don't ask for a token refresh
As the title says, the error doesn't always show for whatever reason. To be honest, it probably shouldn't show, and the token should just be refreshed automatically in the background.

### PR DESCRIPTION
**Pull request by zecakeh**
Refresh the token if it can. If the password has changed, prompt for the new password.

This fixes #67.

Tested on web Firefox Linux
Tested on native Android

### CODE REVIEW

File: src/ErrorBoundary.tsx [hunk 1]
```diff
@@ -42,10 +43,26 @@ function emailDevelopers(error: Error, logs: string | undefined) {
...
-  return (
+  React.useEffect(() => {
+    const fetch = async () => {
+      try {
+        await etebase.fetchToken();
+
+        dispatch(login(etebase));
+        dispatch(popError());
+      } catch (e) {
+        setFetchFailed(true);
```
Comment (tasn) [inline src/ErrorBoundary.tsx:58]: Fetch can fail for more than just Unauthorized. If it fails with Unauthorized we should do this, otherwise we should show a temporary error dialog with the error. I guess it's a rare case anyway though. I'll merge it regardless, but just noting it here for the future.

**Comment by tasn** [comment]
Thank you so much, and great work!

File: src/ErrorBoundary.tsx (hunk 1 above)
Comment (zecakeh) [reply src/ErrorBoundary.tsx:58]: Oh yeah I thought about it for a sec and I forgot later. I think it's a one line fix, just have to `addError` so it goes back to ErrorBoundary. I'll make a quick PR

Comment (tasn) [reply src/ErrorBoundary.tsx:58]: Not sure about that, because it could potentially cause infinite loops. I don't know if/what is the easy solution tbh. Maybe better to just leave it.

Comment (zecakeh) [reply src/ErrorBoundary.tsx:58]: The only case it would cause an infinite loop is if it was an UnauthorizedError, because it's the only error using that dialog, and that's the only case we won't send it back because we prompt for the password.

One question I had, is how does th sync work when you dispatch login. Is it one way server -> client? Because that would make the user lose his local changes.

Comment (tasn) [reply src/ErrorBoundary.tsx:58]: Could you rephrase your question? I don't understand what you mean. Why would it make the user lose his local changes? The login function should only update the stored etebase session.

Comment (zecakeh) [reply src/ErrorBoundary.tsx:58]: When I tested my commit, I noticed that "Pulling changes" appears in the Note List after the password is entered. My question is just: is this a one-way server->client pull or does it also merge local changes?

Comment (tasn) [reply src/ErrorBoundary.tsx:58]: It should just trigger a normal sync.

Comment (zecakeh) [reply src/ErrorBoundary.tsx:58]: Oh OK then :+1:
